src/main/java/com/scaleunlimited/flinkcrawler/topology/CrawlTopologyBuilder.java

```diff
@@ -399,6 +422,26 @@ public FetchUrl map(Tuple3<CrawlStateUrl, FetchUrl, FetchUrl> justHasFetchUrl)
                                 TimeUnit.SECONDS, 10000)
                         .name("FetchUrlsFunction");
 
+        // Save off the content by converting it to a WARC record and passing it on to the provided
+        // content sink function.
+        DataStream<FetchResultUrl> fetchResultUrlsToSave = fetchResultUrls;
+        SingleOutputStreamOperator<Tuple2<NullWritable, WARCWritable>> warcStream = fetchResultUrlsToSave
+                .flatMap(new CreateWARCWritableFunction(_userAgentString))
+                .name("Create WARC writable");
+        
+        DataStreamSink<Tuple2<NullWritable, WARCWritable>> contentSinkUsingOutputFormat = null;
+        if (_contentSink != null) {
+            contentSinkUsingOutputFormat = warcStream.addSink(_contentSink);
+        } else {
+            Job job = Job.getInstance();
+            WARCOutputFormat.setOutputPath(job, new Path(_contentFilePathString));
+            HadoopOutputFormat<NullWritable, WARCWritable> hadoopOutputFormat = new HadoopOutputFormat<NullWritable, WARCWritable>(new WARCOutputFormat(), job);
+            contentSinkUsingOutputFormat = warcStream.writeUsingOutputFormat(hadoopOutputFormat);
+        } 
+
+        contentSinkUsingOutputFormat
```

Comment:
No need to ensure the same parallelism as the ParseFunction that generates the output?

Reply:
I am not sure... Since you are asking, is that what you would expect ?

Reply:
That's the way it was set up before. Please check with Ken.

Reply:
You'd want to generate the .dot file and visualize it, to ensure that we don't get extra (unnecessary) steps to re-partition data to adjust for different parallelism.